### PR TITLE
Replace ammonia lib with manual html escaping (fixes #3774)

### LINKED ISSUE #3774: [Bug]: `<` is incorrectly sanitized within code blocks
### Requirements

- [X] Is this a bug report? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a single bug? Do not put multiple bugs in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Summary

When writing a comment like this:

````md
`type` can be used with generic types like `Result<Success, Error>`.

```rust
pub type LibResult = Result<Success, Error>;
```
````

This is "sanitized" to the following:

````md
`type` can be used with generic types like `Result&lt;Success, Error>`.

```rust
pub type LibResult = Result;
```
````

TL;DR: `<` in inline code is replaced with `&lt;`, and generics (which look like an HTML tag) are removed completely.

**NOTE: Sanitization is often discouraged, because it is error prone, as can be observed here. Instead, use a markdown renderer that correctly handles arbitrary, untrusted input!** Most Markdown rendering libraries already disallow HTML, or can be configured to do so. For example, `markdown-it` (used by lemmy-ui) disables HTML by default.

Alternatively, the generated HTML can be sanitized in the client _after converting the markdown to HTML_.

### Steps to Reproduce

1. Write a comment with `<` in inline code or an HTML tag in a code block
2. Submit the comment

### Technical Details

N/A

### Version

0.18.3

### Lemmy Instance URL

https://programming.dev

### PR DESCRIPTION
**Pull request by Nutomic**
The ammonia library is not what we need, for our use case its better to replace problematic HTML characters directly. This should fix all the problems like removing <script> tags from code blocks. Based on my testing, lemmy-ui currently does some html escaping of its own, so we need to get rid of that.

### CODE REVIEW

File: crates/api_common/src/utils.rs [hunk 1]
```diff
@@ -798,29 +798,43 @@ pub fn generate_moderators_url(community_id: &DbUrl) -> Result<DbUrl, LemmyError
...
+pub fn sanitize_html_federation(data: &str) -> String {
+  data
+    .replace("<", "&lt;")
+    .replace("\"", "&quot;")
+    .replace("'", "&#x27;")
+}
+
+pub fn sanitize_html_federation_opt(data: &Option<String>) -> Option<String> {
+  data.as_ref().map(|d| sanitize_html_federation(d))
 }
```
Comment (Nutomic) [inline crates/api_common/src/utils.rs:830]: Main change here.

**Review by dessalines — APPROVED** [review]
👍

**Review by dessalines — APPROVED** [review]

**Comment by AndrejsK** [comment]
Looks like this PR has accidentally removed a table from the diesel schema added in #3927

**Comment by dessalines** [comment]
I've added it back in, in the scaled-sort pr.

**Pull request merged**